lib/wibox/layout/align.lua

```diff
@@ -35,127 +35,336 @@ local base = require("wibox.widget.base")
-                --  widget to take up any space (as the first if block is skipped
-                --  if this is the case)
-                if self._private.expand == "inside" or not self._private.second then
-                    size_remains = size_remains - h
-                end
-            else
-                w, _ = base.fit_widget(self, context, self._private.first, size_remains, height)
-                size_first = w
-                if self._private.expand == "inside" or not self._private.second then
-                    size_remains = size_remains - w
-                end
-            end
+
+    -- Wrapper around layout_place_widget function to pass less parameters
+    local function place_widget(widget, offset, size)
+        -- Can't place a widget if it does not exist or there is no more size
+        if not widget or size_remains <= 0 then
+            return
+        end
+        -- Default values
+        if not size then
+            size = fit_widget(widget, size_remains)
         else
-            if self._private.dir == "y" then
-                h = size_remains
-            else
-                w = size_remains
-            end
+            fit_widget(widget, size)
         end
-        table.insert(result, base.place_widget_at(self._private.first, 0, 0, w, h))
+
+        local placed = layout_place_widget(
+            self, width, height,
+            widget, offset, size
+        )
+        table.insert(result, placed)
+        size_remains = size_remains - size
     end
-    -- size_remains will be <= 0 if first used all the space
-    if self._private.third and size_remains > 0 then
-        local w, h, _ = width, height, nil
-        if self._private.expand ~= "outside" then
-            if self._private.dir == "y" then
-                _, h = base.fit_widget(self, context, self._private.third, width, size_remains)
-                -- give the middle widget the rest of the space for "inside" mode
-                if self._private.expand == "inside" then
-                    size_remains = size_remains - h
-                end
-            else
-                w, _ = base.fit_widget(self, context, self._private.third, size_remains, height)
-                if self._private.expand == "inside" then
-                    size_remains = size_remains - w
-                end
-            end
+
+    -- Size of the layout before any placements
+    local size_total = size_remains
+
+    if self._private.expand == align.expand_modes.NONE then
+        local size_second = fit_widget(second)
+
+        if size_second >= size_total then
+            -- Second widget takes all layout's space, no need to place another widgets
+            place_widget(second, 0, size_total)
+        elseif size_second == 0 then
+            -- There is no second widget
+            local size_side = floor(size_total / 2)
+
+            local size_first = fit_widget(first, size_side)
```

Comment:
nit: Since it's used only once and the variable name doesn't provide more detail than what we can guess with the function name/parameter name, you can remove the variable and call fit_widget directly in the place_widget invocation.

Reply:
OK, I will remove this variable.